Add `visualize.py` CLI

rahularya50 opened the pull request:
Usage documented in README. Tested as
```
make visualize VIZ_INSTANCE=../../sp22-170-proj/cases/simple/simple-1.in VIZ_OUTPUT=output.svg VIZ_SOLUTION=../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```
as well as
```
python3 visualize.py ../../sp22-170-proj/cases/simple/simple-1.in test.svg --with-solution ../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```

Also tested omitting the solution, the `VIZ_OUTPUT` variable, and invalid arguments.

Question: do we want a Makefile-based way of doing this at all, or just tell them to use the python script? I'm not entirely clear what commands should go in the Makefile, and what should just be scripts.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #10
* #9
* #8
* #1

rahularya50 committed:
Add `make visualize` command

[ghstack-poisoned]

rahularya50 committed:
Update on "Add `make visualize` command"




[ghstack-poisoned]

rahularya50 committed:
Update on "Add `make visualize` command"




[ghstack-poisoned]

rahularya50 committed:
Update on "Add `make visualize` command"




[ghstack-poisoned]

tylerhou commented on python/visualize.py:
```diff
@@ -10,3 +14,47 @@ class VisualizationConfig:
+    else:
+        svg = instance.visualize_as_svg(config)
+
+    if args.output == "-":
+        print(svg)
+    else:
+        with open(args.output, "w") as f:
+            f.write(str(svg))
```
Could use `StdoutFileWrapper` here

rahularya50 committed:
Update on "Add `make visualize` command"

Usage documented in README. Tested as
```
make visualize VIZ_INSTANCE=../../sp22-170-proj/cases/simple/simple-1.in VIZ_OUTPUT=output.svg VIZ_SOLUTION=../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```
as well as
```
python3 visualize.py ../../sp22-170-proj/cases/simple/simple-1.in test.svg --with-solution ../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```

Also tested omitting the solution, the `VIZ_OUTPUT` variable, and invalid arguments.

Question: do we want a Makefile-based way of doing this at all, or just tell them to use the python script? I'm not entirely clear what commands should go in the Makefile, and what should just be scripts.




[ghstack-poisoned]

tylerhou commented:
> Question: do we want a Makefile-based way of doing this at all, or just tell them to use the python script? I'm not entirely clear what commands should go in the Makefile, and what should just be scripts.

I think telling the to use the Python script would be fine

rahularya50 committed:
address review comments on "Add `make visualize` command"

Usage documented in README. Tested as
```
make visualize VIZ_INSTANCE=../../sp22-170-proj/cases/simple/simple-1.in VIZ_OUTPUT=output.svg VIZ_SOLUTION=../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```
as well as
```
python3 visualize.py ../../sp22-170-proj/cases/simple/simple-1.in test.svg --with-solution ../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```

Also tested omitting the solution, the `VIZ_OUTPUT` variable, and invalid arguments.

Question: do we want a Makefile-based way of doing this at all, or just tell them to use the python script? I'm not entirely clear what commands should go in the Makefile, and what should just be scripts.




[ghstack-poisoned]

rahularya50 committed:
update readme on "Add `make visualize` command"

Usage documented in README. Tested as
```
make visualize VIZ_INSTANCE=../../sp22-170-proj/cases/simple/simple-1.in VIZ_OUTPUT=output.svg VIZ_SOLUTION=../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```
as well as
```
python3 visualize.py ../../sp22-170-proj/cases/simple/simple-1.in test.svg --with-solution ../../sp22-170-proj/cases/simple/simple-1-annealing.out 
```

Also tested omitting the solution, the `VIZ_OUTPUT` variable, and invalid arguments.

Question: do we want a Makefile-based way of doing this at all, or just tell them to use the python script? I'm not entirely clear what commands should go in the Makefile, and what should just be scripts.




[ghstack-poisoned]
